blocks/bricks/recurrent.py

```diff
@@ -300,6 +301,109 @@ def apply(self, input_=None, state=None, mask=None):
         return next_state
 
 
+class LSTM(BaseRecurrent, Initializable):
+    u"""Long Short Term Memory.
+
+    Every unit of an LSTM is equipped with input, forget and output gates.
+    This implementation aims to do as many computations in parallel as
+    possible and expects the last dimension of the input to be four times
+    the output dimension.
+
+    Parameters
+    ----------
+    dim : int
+        The dimension of the hidden state.
+
+    Notes
+    -----
+    See :class:`.Initializable` for initialization parameters.
+
+    """
+    @lazy
+    def __init__(self, dim, **kwargs):
+        super(LSTM, self).__init__(**kwargs)
+        self.dim = dim
+
+    def get_dim(self, name):
+        if name == 'inputs':
+            return self.dim * 4
+        else:
```

Comment:
This is a bit dangerous, because it means that if a person requests a non-existing `name` he will still get a return value. It's best to check for specific names here (and I think @rizar has always given `mask` a dimension of ) e.g. 
```python
if name == 'mask':
    return 0
if name in ['states', 'cells']:
    return self.dim
return super(LSTM, self).get_dim(name)  # This raises an informative error
```